crates/sequencer/src/state.rs

```diff
@@ -143,6 +170,27 @@ impl BlockifierStateReader for &mut State {
     }
 }
 
+impl From<&SerializableState> for State {
+    fn from(serializable_state: &SerializableState) -> Self {
+        let mut state = Self {
+            classes: FxHashMap::default(),
+            compiled_class_hashes: serializable_state.compiled_classes_hash.clone(),
+            contracts: serializable_state.contracts.clone(),
+            storage: serializable_state.storage.clone(),
+            nonces: serializable_state.nonces.clone(),
+        };
+
+        serializable_state.classes.iter().for_each(|(class_hash, contract_class)|{
+           state.classes.insert(
+               *class_hash,
+               contract_class.class.clone().try_into().unwrap_or_else(|error| panic!("failed to convert SerializableClassRecord to ContractClass for class_hash {},\n error {}", class_hash, error))
```

Comment:
must we panic here?
can't we implement TryFrom and return a result?

Reply:
I think panic here is fine right? Because this situation is a bug, it means that there is an error in our serialisation for contract class { which is already tested by dojo! }, and from my understanding, panics are reserved for bugs, hence, from our assumption, this should never happen!

Reply:
I think we should never have a `panic!` in our code, would make a lot more sense to rely the error upstream if we have one